warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -3229,8 +3259,254 @@ public static Date getEndDateForIndexLookup(Date endDate) {
 
     @Override
     public void finalize() {
-        if (null != builderThread) {
-            builderThread.shutdown();
+        if (null != executor) {
+            executor.shutdown();
         }
     }
+
+    protected CompositeMetadata getCompositeMetadata() {
+        if (compositeMetadata == null && compositeMetadataCallable != null) {
+            TraceStopwatch stopwatch = stageStopWatch.newStartedStopwatch(compositeMetadataCallable.stageName());
+            try {
+                while (compositeMetadata == null) {
+                    compositeMetadata = compositeMetadataFuture.get(concurrentTimeoutMillis, TimeUnit.MILLISECONDS);
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                log.error("Failed to fetch CompositeMetadata");
+                log.error(e.getStackTrace());
+                throw new RuntimeException(e);
```

Comment:
Maybe throw a specialty exception so we can track how often this leads to a query failing? This could be caught internally or returned with a new datawave error code

Reply:
done in next commit